### PR TITLE
refactor: electron to typescript

### LINKED ISSUE #165: ️🛠 - Refactor electron libs to typescript
### Task description

We need to improve our code quality to have better maintainability and be easier to port our application to Tauri if we want to

### Requirements

TBD

### Creation checklist

- [X] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by jeeanribeiro**
## Summary

closes #165 

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [x] I have followed the contribution guidelines for this project
-   [x] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by maxwellmattryan — DISMISSED** [review]
LGTM :cherry_blossom:

**Review by Tuditi — CHANGES_REQUESTED** [review]
I cannot create a new profile (error 1)/enter a pin due to the following errors on a fresh build:
I think the methods of the PincodeManager have to be bound across the context bridge as well.
Error 1:
```
core_app__WEBPACK_IMPORTED_MODULE_1__.Platform.PincodeManager.set is not a function
    at initialisePincodeManager.ts:11:1
    at Generator.next (<anonymous>)
    at initialiseOnboardingProfile.ts:28:1
    at new Promise (<anonymous>)
    at __async (initialiseOnboardingProfile.ts:28:1)
    at initialisePincodeManager (initialisePincodeManager.ts:10:1)
    at handleSetPin (EnterPinView.svelte:33:1)
    at HTMLFormElement.onSetPinClick (EnterPinView.svelte:46:1)
    at HTMLFormElement.<anonymous> (index.mjs:432:1)
```

Error 2:
```
TypeError: _core_app__WEBPACK_IMPORTED_MODULE_3__.Platform.PincodeManager.verify is not a function
    at onSubmit (EnterPinView.svelte:95:1)
    at $$self.$$.update (EnterPinView.svelte:49:1)
    at update (index.mjs:1294:1)
    at flush (index.mjs:1258:1)
```

**Comment by maxwellmattryan** [comment]
@Tuditi Did you pull in the latest changes when you had those errors?

File: packages/desktop/lib/electron/managers/keychain.manager.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export default class KeychainManager {
+    private serviceName: string
+
+    constructor() {
+        this.serviceName = app?.isPackaged ? app.getName() : 'Bloom — Dev'
+    }
+
+    public getAll(): Promise<Credential[]> {
+        return keytar.findCredentials(this.serviceName)
+    }
```
Comment (Tuditi) [inline packages/desktop/lib/electron/managers/keychain.manager.ts]: We can remove this, since we're never emitting e `get-all` event.

**Review by Tuditi — CHANGES_REQUESTED** [review]
Yup, I forgot to pull. Everything seems to work now. One minor thing to be changed.

**Review by maxwellmattryan — DISMISSED** [review]
LGTM :cherry_blossom:

**Review by Tuditi — APPROVED** [review]
I removed the 'getAll' event handler myself :relaxed: . LGTM!

**Pull request merged**